kolibri/plugins/coach/assets/src/state/store.js

```diff
@@ -54,15 +54,26 @@ const mutations = {
   SET_MODAL(state, modalName) {
     state.pageState.modalShown = modalName;
   },
+
   ADD_GROUP(state, group) {
     state.pageState.groups.push(group);
   },
-  UPDATE_GROUP(state, groupId, updatedGroup) {
-    state.pageState.groups.forEach((group, index, arr) => {
-      if (group.id === groupId) {
-        arr[index] = updatedGroup;
-      }
-    });
+  RENAME_GROUP(state, groupId, newGroupName) {
```

Comment:
I get the feeling these might be a little too complex for the store. Personally, I expect most of the logic to be taken care of inside of the `Actions` file. I know this is how we're doing it inside of `Management` --  I'll fix that one day.